src/main/java/mate/academy/spring/controller/MovieSessionController.java

```diff
@@ -0,0 +1,61 @@
+package mate.academy.spring.controller;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.spring.dto.MovieSessionRequestDto;
+import mate.academy.spring.dto.MovieSessionResponseDto;
+import mate.academy.spring.model.MovieSession;
+import mate.academy.spring.service.MovieSessionService;
+import mate.academy.spring.service.mapping.MovieSessionDtoMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+@RequestMapping("/movie-sessions")
+public class MovieSessionController {
+    @Autowired
+    private MovieSessionService movieSessionService;
+    @Autowired
+    private MovieSessionDtoMapper movieSessionDtoMapper;
+
+    @PostMapping
+    public MovieSessionResponseDto create(
+            @RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        return movieSessionDtoMapper.toDto(
+                movieSessionService.add(
+                        movieSessionDtoMapper.toModel(movieSessionRequestDto)));
+    }
+
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> findAvailableSessions(
+            @RequestParam Long movieId,
+            @RequestParam @DateTimeFormat(pattern = "dd.MM.yyyy") LocalDate date) {
+        return movieSessionService.findAvailableSessions(movieId, date).stream()
+                .map(movieSessionDtoMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @PutMapping("/{id}")
+    public MovieSessionResponseDto update(
+            @PathVariable Long id, @PathVariable MovieSessionRequestDto movieSessionRequestDto) {
+        MovieSession movieSession = movieSessionDtoMapper.toModel(movieSessionRequestDto);
+        movieSession.setId(id);
+        MovieSession updatedMovieSession = movieSessionService.update(movieSession);
+        return movieSessionDtoMapper.toDto(updatedMovieSession);
+    }
+
+    @DeleteMapping("/{id}")
+    public void delete(
+            @PathVariable Long id,
+            @PathVariable MovieSessionRequestDto movieSessionRequestDto) {
```

Comment:
```suggestion
            ) {
```
You don't use this parameter